esp/platform/espp.cpp

```diff
@@ -354,6 +358,91 @@ static void usage()
 
 IPropertyTree *buildApplicationLegacyConfig(const char *application, const char* argv[]);
 
+// Modified version of jlib's loadTraceFlags. The modification adds special handling for traceLevel.
+//
+// Using loadTraceFlags, `traceStandard: true`, `traceStandard: false`, and `traceStandard: maybe`
+// are non-intuitively equivalent. The property name is the only thing that matters, with the
+// value being ignored.
+//
+// Also using loadTraceFlags, all defined trace levels may be included in one configuration. The
+// final, accepted, level is determined by the order of `optNames` list items. An order optimized
+// for elevating the level cannot be used in reverse. While a configuration with multiple properties
+// should be discouraged, using `helm install --set` to change the level will add a new property
+// rather than changing the existing one.
+//
+// With loadEspTraceFlags, `traceLevel` is the only property that sets the trace level. The value
+// is both relevant and intuitive, and can be overridden with `helm install --set`.
+static TraceFlags loadEspTraceFlags(const IPropertyTree *ptree, const std::initializer_list<TraceOption> &optNames, TraceFlags dft)
+{
+    for (const TraceOption& option: optNames)
+    {
+        VStringBuffer attrName("@%s", option.name);
+        const char* value = nullptr;
+        if (!(value = ptree->queryProp(attrName)))
+        {
+            attrName.setCharAt(0, '_');
+            if (!(value = ptree->queryProp(attrName)))
+                continue;
+        }
+        if (streq(value, "default")) // allow a configuration to explicitly request a default value
+            continue;
+        if (streq(option.name, propTraceLevel)) // non-Boolean traceLevel
```

Comment:
This logic makes sense, but would probably make more sense added to loadTraceFlags.

I would add add an extra enumeration traceDetail[Level?] = 0xffffffff, and check that value with this processing:

if (option.value == traceDetail)...

(I am suggesting traceDetail rather than traceLevel to avoid clashes with the existing traceLevel option which has different ranges of values?)

Reply:
Agree on loadTraceFlags. With multiple comments here, suggest we fix this here, then it will be easy to lift and shift into jlib.

Reply:
It would replace the comparison on the name.